src/ControlSystem/Tags.hpp

```diff
@@ -175,17 +249,18 @@ struct MeasurementTimescales : db::SimpleTag {
 /// This struct collects all the options for a given control system during
 /// option parsing. Then during initialization, the options can be retrieved via
 /// their public member names and assigned to their corresponding DataBox tags.
-template <size_t DerivOrder>
+template <typename ControlSystem>
 struct OptionHolder {
+  static constexpr size_t deriv_order = ControlSystem::deriv_order;
```

Comment:
Here, for instance, would be a good place for an `assert_conforms_to`, before this line